samtranslator/model/sam_resources.py

```diff
@@ -2331,11 +2332,11 @@ def _parse_datasource_role(
 
         # If the user doesn't have their own role, then we will create for them if TableArn is defined.
         table_arn = sam_expect(
-            ddb_datasource.TableArn, relative_id, f"DynamoDBDataSources.{relative_id}.TableArn"
```

Comment:
Shouldn't this stay as `relative_id`? From customer's perspective the "path" is unchanged.

Reply:
Customer should be able to reference DS created inside SAM resource in an AppSync resource like this 
```
!GetAtt SamGraphQl.DataSource.NotesReadDataSource.Name
```
where `NotesReadDataSource` is `relative_id`. It might be more logical indeed if it will be the same as the part of arn.

Reply:
I think he's talking about the error message will point to a path that is not necessarily correct since it's using the logical ID now. Agreed, I'll revert this back to `relative_id` so its the same as in SAM template.